packages/fx-core/src/core/middleware/consolidateLocalRemote.ts

```diff
@@ -0,0 +1,335 @@
+import * as util from "util";
+import { ManifestTemplate } from "../../plugins/resource/spfx/utils/constants";
+
+const upgradeButton = "Upgrade";
+let userCancelFlag = false;
+const backupFolder = ".backup";
+
+export const ProjectConsolidateMW: Middleware = async (
+  ctx: CoreHookContext,
+  next: NextFunction
+) => {
+  if (await needMigrateToArmAndMultiEnv(ctx)) {
+    next();
+  } else if ((await needConsolidateLocalRemote(ctx)) && checkMethod(ctx)) {
+    sendTelemetryEvent(Component.core, TelemetryEvent.ProjectMigratorNotificationStart);
+    const res = await TOOLS?.ui.showMessage(
+      "warn",
+      getLocalizedString("core.consolidateLocalRemote.Message"),
+      true,
+      upgradeButton
+    );
+    const answer = res?.isOk() ? res.value : undefined;
+    if (!answer || answer != upgradeButton) {
+      sendTelemetryEvent(Component.core, TelemetryEvent.ProjectConsolidateNotification, {
+        [TelemetryProperty.Status]: ProjectMigratorStatus.Cancel,
+      });
+      ctx.result = err(ConsolidateCanceledError());
+      outputCancelMessage(ctx);
+      return;
+    }
+    sendTelemetryEvent(Component.core, TelemetryEvent.ProjectConsolidateNotification, {
+      [TelemetryProperty.Status]: ProjectMigratorStatus.OK,
+    });
+
+    try {
+      await consolidateLocalRemote(ctx);
+      await next();
+    } catch (error) {
+      sendTelemetryErrorEvent(
+        Component.core,
+        TelemetryEvent.ProjectConsolidateError,
+        assembleError(error, CoreSource)
+      );
+      throw error;
+    }
+  } else {
+    await next();
+  }
+};
+
+// check if config.local.json and manifest.template.json exist
+async function needConsolidateLocalRemote(ctx: CoreHookContext): Promise<boolean> {
+  if (!isConfigUnifyEnabled()) {
+    return false;
+  }
+  const inputs = ctx.arguments[ctx.arguments.length - 1] as Inputs;
+  if (!inputs.projectPath) {
+    return false;
+  }
+  const fxExist = await fs.pathExists(path.join(inputs.projectPath as string, ".fx"));
+  if (!fxExist) {
+    return false;
+  }
+
+  const localEnvExist = await fs.pathExists(
+    path.join(inputs.projectPath as string, ".fx", "configs", "config.local.json")
+  );
+  const consolidateManifestExist = await fs.pathExists(
+    path.join(inputs.projectPath as string, "templates", "appPackage", "manifest.template.json")
+  );
+  if (!localEnvExist && !consolidateManifestExist) {
+    return true;
+  }
+  return false;
+}
+
+function outputCancelMessage(ctx: CoreHookContext) {
+  TOOLS?.logProvider.warning(`[core] Upgrade cancelled.`);
+
+  const inputs = ctx.arguments[ctx.arguments.length - 1] as Inputs;
+  if (inputs.platform === Platform.VSCode) {
+    TOOLS?.logProvider.warning(
+      `[core] Notice upgrade to new configuration files is a must-have to continue to use current version Teams Toolkit. If you are not ready to upgrade and want to continue to use the old version Teams Toolkit, please find Teams Toolkit in Extension and install the version <= ????`
```

Comment:
do we need to move these warning message to `package.nls.json`?

Reply:
For output channel, we do not need to move it to ```package.nls.json```.